jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -602,6 +612,8 @@ public static String decodePath(String path, int offset, int length, BiConsumer<
         {
             if (LOG.isDebugEnabled())
                 LOG.debug("{} {}", path.substring(offset, offset + length), e.toString());
+            if (!fallbackToIso88591)
+                throw new IllegalArgumentException("Not UTF-8, not decoding in ISO-8859-1", e);
```

Comment:
I also don't think we need this... unless an illegal UTF8 character can be decoded as `/` in iso8859-1, which I don't believe is possible... but even if it were, then we'd leave the `/` encoded and there would be nothing ambigous.

Reply:
if anything, I'm dubious that we should ever fall back to iso8859-1